src/lib/dnssd/Resolver_ImplMinimalMdns.cpp

```diff
@@ -452,18 +453,46 @@ void MinMdnsResolver::AdvancePendingResolverStates()
         else if (resolver->IsActiveOperationalParse())
         {
             MATTER_TRACE_SCOPE("Active operational delegate call", "MinMdnsResolver");
-            ResolvedNodeData nodeData;
+            ResolvedNodeData nodeResolvedData;
+            CHIP_ERROR err = CHIP_ERROR_UNINITIALIZED;
+
+            err = resolver->Take(nodeResolvedData);
+
+            if (mActiveResolves.HasBrowseFor(chip::Dnssd::DiscoveryType::kOperational))
+            {
+                DiscoveredNodeData nodeData;
+                if (err == CHIP_NO_ERROR)
+                {
+                    OperationalNodeBrowseData opNodeData;
+                    opNodeData.peerId     = nodeResolvedData.operationalData.peerId;
+                    opNodeData.hasZeroTTL = nodeResolvedData.operationalData.hasZeroTTL;
+                    nodeData.Set<OperationalNodeBrowseData>(opNodeData);
+                }
+                if (err != CHIP_NO_ERROR)
+                {
+                    ChipLogError(Discovery, "Failed to take discovery result: %" CHIP_ERROR_FORMAT, err.Format());
+                }
+                else if (mDiscoveryContext != nullptr)
+                {
+                    mDiscoveryContext->OnNodeDiscovered(nodeData);
+                }
+                else
+                {
+#if CHIP_MINMDNS_HIGH_VERBOSITY
+                    ChipLogError(Discovery, "No delegate to report operational node discovery");
+#endif
+                }
+            }
 
-            CHIP_ERROR err = resolver->Take(nodeData);
             if (err != CHIP_NO_ERROR)
             {
-                ChipLogError(Discovery, "Failed to take discovery result: %" CHIP_ERROR_FORMAT, err.Format());
+                ChipLogError(Discovery, "Failed to take ResolvedNodeData - result: %" CHIP_ERROR_FORMAT, err.Format());
```

Comment:
So if Take() fails we will log the error twice if `HasBrowseFor(chip::Dnssd::DiscoveryType::kOperational)`?  Seem confusing.

Reply:
thought it would be good idea to have separate error logs for browse/discovery and resolution.

Reply:
I think my point is that we will log both `Failed to take discovery result` and `Failed to take ResolvedNodeData` when `DiscoveryType::kOperational`.

Maybe this second `if (err != CHIP_NO_ERROR)` should just be `else if` so it does not get hit when `DiscoveryType::kOperational`?

Reply:
I suggest having a single log just after `Take` is executed. Also, after rebasing to https://github.com/project-chip/connectedhomeip/commit/881676de173e779b9471eaae944c33e5780b038c, the code will continue to the next record in such a case, which it should do.

Reply:
Moved log just after Take() as suggested.